main_test.go

```diff
@@ -3,3 +3,484 @@
 // 📝 Github Repository: https://github.com/gofiber/fiber
 
 package keyauth
+
+import (
+	"strings"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+func validateAPIKey(ctx *fiber.Ctx, s string) (bool, error) {
+	if s == "" {
+	  return false, &fiber.Error{Code: 403, Message: "Missing API key"}
+	}
+	if s == "valid-key" {
+	  return true, nil
+	}
+	return false, &fiber.Error{Code: 403, Message: "Invalid API key"}
+}
+
+func TestKeyLookupHeader(t *testing.T) {
+	// Initialize a Fiber app
+	app := fiber.New()
+
+	// Use the KeyAuth middleware with the default configuration
+	app.Use(New())
+
+	// Define a test handler
+	app.Get("/", func(c *fiber.Ctx) error {
+		return c.SendString("API key is valid")
+	})
+
+	// Test the default KeyLookup value "header:Authorization"
+	// Create a request without an API key and  Send the request to the app
+	res, err := app.Test(httptest.NewRequest("GET", "/", nil))
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Read the response body into a string
+	body, _ := ioutil.ReadAll(res.Body)
+
+	// Check that the response has the expected status code and body
+	if res.StatusCode != http.StatusBadRequest {
```

Comment:
Using utils.AssertEqual seems cleaner to me

Reply:
Agree! Done